cmd/agent/main.go

```diff
@@ -1,3 +1,47 @@
 package main
 
-func main() {}
+import (
+	"context"
+	"github.com/koyif/metrics/internal/agent"
+	"github.com/koyif/metrics/internal/agent/client"
+	"github.com/koyif/metrics/internal/agent/config"
+	"github.com/koyif/metrics/internal/agent/scraper"
+	"github.com/koyif/metrics/internal/agent/storage"
+	"log/slog"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+func main() {
+	cfg := config.Load()
+
+	if err := run(cfg); err != nil {
+		panic(err)
+	}
+}
+
+func run(cfg *config.Config) error {
+	stg := storage.New()
+	sc := scraper.New(stg)
+	cl, err := client.New(cfg, &http.Client{})
+	if err != nil {
+		return err
```

Comment:
Ошибки из сторонних пакетов лучше оборачивать в дополнительный контекст:

```suggestion
		return fmt.Errorf("failed to initialize a metrics client: %w", err)
```

Reply:
Это не сторонняя библиотека. client.New - это мой клиент. А внутри ошибка обернута так:
`fmt.Errorf("%s: error creating MetricsClient: %w", op, err)`